satellite/orders/orders_test.go

```diff
@@ -22,17 +22,64 @@ import (
 	"storj.io/storj/satellite"
 	"storj.io/storj/satellite/orders"
 	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+	"storj.io/storj/storagenode"
 	"storj.io/storj/uplink"
 )
 
+func TestCleanArchive(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			StorageNode: func(index int, config *storagenode.Config) {
+				config.Storage2.Orders.CleanupInterval = 100 * time.Millisecond
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		planet.Satellites[0].Audit.Service.Loop.Stop()
+		node := planet.StorageNodes[0]
+		service := node.Storage2.Orders
+		service.Sender.Pause()
+		service.Cleanup.Pause()
+
+		expectedData := testrand.Bytes(50 * memory.KiB)
+
+		redundancy := noLongTailRedundancy(planet)
+		err := planet.Uplinks[0].UploadWithConfig(ctx, planet.Satellites[0], &redundancy, "testbucket", "test/path", expectedData)
+		require.NoError(t, err)
+
+		service.Sender.TriggerWait()
+
+		// let archived order age for 100 milliseconds
+		time.Sleep(100 * time.Millisecond)
+
+		err = planet.Uplinks[0].UploadWithConfig(ctx, planet.Satellites[0], &redundancy, "testbucket", "test/path", expectedData)
+		require.NoError(t, err)
+
+		service.Sender.TriggerWait()
```

Comment:
Do you need to sleep again here to prevent flappiness? In general sleeps in tests make me nervous, but I want to make sure we reduce the risk of a flappy test as much as possible.

Reply:
I also was nervous about adding sleep, but you can't manually set an order's `archived_at`: it's set to time.Now() inside the Archive function. So, I'm not sure how else to test that deletion based on the time of archival works without sleeping for some period

Reply:
It's usually better to make a way to pass the desired time into the function instead of doing time.Now(), and if that also avoids a sleep call in a test, that makes it even better!